mbtiles/src/bin/mbtiles.rs

```diff
@@ -332,6 +385,216 @@ async fn meta_set_value(file: &Path, key: &str, value: Option<&str>) -> MbtResul
     }
 }
 
+async fn pack(
+    input_directory: &Path,
+    output_file: &Path,
+    scheme: TileScheme,
+) -> anyhow::Result<()> {
+    if !input_directory.exists() {
+        anyhow::bail!(
+            "Input directory does not exist: {}",
+            input_directory.display()
+        );
+    }
+    if !input_directory.is_dir() {
+        anyhow::bail!(
+            "Input path is not a directory: {}",
+            input_directory.display()
+        );
+    }
+
+    let mbt = Mbtiles::new(output_file)?;
+    let mut conn = mbt.open_or_new().await?;
+
+    create_metadata_table(&mut conn).await?;
+    create_flat_tables(&mut conn).await?;
+
+    let walker = WalkDir::new(input_directory);
+    let entries = walker.into_iter().filter_entry(|entry| {
+        let should_include = if entry.file_type().is_dir() {
+            // skip directories except the root unless they have numeric names
+            entry.depth() == 0
+                || entry
+                    .file_name()
+                    .to_str()
+                    .is_some_and(|s| s.parse::<u32>().is_ok())
+        } else {
+            // skip files that do not have a numeric basename
+            entry
+                .file_name()
+                .to_str()
+                .and_then(|s| s.split('.').next().map(|b| b.parse::<u32>().is_ok()))
+                .unwrap_or(false)
+        };
+
+        if !should_include {
+            log::info!(
+                "Skipping {}{}",
+                entry.path().display(),
+                if entry.file_type().is_dir() { "/" } else { "" }
+            );
+        }
+
+        should_include
+    });
+
+    let mut format: Option<String> = None;
+    let mut compress = false;
+
+    for entry in entries {
+        let Some(entry) = entry.ok() else {
+            continue;
+        };
+
+        let path_components: Vec<_> = entry.path().iter().skip(1).collect();
+        let coords: Vec<u32> = path_components
+            .iter()
+            .filter_map(|c| {
+                c.to_str()
+                    .and_then(|s| s.split('.').next())
+                    .and_then(|basename| basename.parse().ok())
+            })
+            .collect();
+
+        if let [z, x, y] = coords.as_slice() {
+            let (z, x, y) = (u8::try_from(*z)?, *x, *y);
+            // TODO: set metadata format from extension of first file, and check that
+            // subsequent files have the same extension
+            if format.is_none() {
+                format = match entry.path().extension().and_then(|s| s.to_str()) {
+                    Some("pbf" | "mvt") => Some("pbf".to_string()),
+                    Some("jpg" | "jpeg") => Some("jpg".to_string()),
+                    Some("webp") => Some("webp".to_string()),
+                    Some("png") => Some("png".to_string()),
```

Comment:
you probably want to use https://github.com/maplibre/martin/blob/6e870e510725064267cadda415ff4a55fdbe115f/martin-tile-utils/src/lib.rs#L86-L89 to convert extension -> format -> back to format value. This way you will also support future formats